### PR TITLE
Vsphere ccm support

### PR DESCRIPTION
**Pull request by lobziik**
Add bits for deploy ccm on vsphere platform

Successfully migrated OCP cluster on my lab vSphere once.

### CODE REVIEW

**Review by Fedosin — APPROVED** [review]
/approve

File: pkg/cloud/vsphere/assets/cloud-controller-manager-daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,118 @@
+kind: DaemonSet
```
Comment (JoelSpeed) [inline pkg/cloud/vsphere/assets/cloud-controller-manager-daemonset.yaml:1]: Please use a Deployment to be consistent with other providers

**Review by JoelSpeed — COMMENTED** [review]
I think it's better to be consistent with the other platforms and use a Deployment.
In some scenarios there will be 5 control plane nodes, we don't need 5 CCM pods sat there with 4 doing nothing. A deployment with 2 replicas should suffice

File: pkg/cloud/vsphere/assets/cloud-controller-manager-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+              if [[ -f /etc/kubernetes/apiserver-url.env ]]; then
+                source /etc/kubernetes/apiserver-url.env
+              fi
+              exec /bin/vsphere-cloud-controller-manager \
+                --v=3 \
+                --cloud-config=$(CLOUD_CONFIG) \
+                --cloud-provider=vsphere \
+                --controllers=* \
+                --configure-cloud-routes=false \
+                --bind-address=127.0.0.1 \
```
Comment (JoelSpeed) [inline pkg/cloud/vsphere/assets/cloud-controller-manager-deployment.yaml:86]: Does this prevent the pod being scraped for metrics?

**Review by Fedosin — APPROVED** [review]
/lgtm

**Pull request merged**